src/test/java/com/puppycrawl/tools/checkstyle/AbstractPathTestSupport.java

```diff
@@ -79,14 +78,15 @@ protected final String getResourcePath(String filename) {
 
     /**
      * Reads the contents of a file.
+     * IDEA inspection RedundantThrows suppressed as a false positive.
```

Comment:
This is not really appropriate to place into the main Javadoc documentation. Can we not attach this to the `noinspection`?

Reply:
This is how we always do this.
We do not know now to record reasons of suppression

Reply:
TC won't allow us to do:
> `@noinspection RedundantThrows - False positive.`

Reply:
TC won't allow us to do

No. Unfortunately.

>`@tcReason`

It triggers some another  violation. We can investigate it in separate issue one more time.

Reply:
"mvn site" is failing:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.10.0:site 
(default-site) on project checkstyle: Error generating maven-javadoc-plugin:3.3.1:javadoc report: 
[ERROR] Exit code: 1 - /home/rivanov/java/github/romani/checkstyle/src/main/java/
com/puppycrawl/tools/checkstyle/Main.java:631: error: unknown tag: noinspectionreason
[ERROR]          * @noinspectionreason: CanBeFinal - we use picocli and it use
[ERROR]            ^
[ERROR] 
[ERROR] Command line was: /opt/jvm/jdk1.8.0_162/jre/../bin/javadoc @options @packages

```

Reply:
new issue - https://github.com/checkstyle/checkstyle/issues/11277

we currently update javadoc on Test classes, nobody read this, bigger  problem will be addressed in separate issue.